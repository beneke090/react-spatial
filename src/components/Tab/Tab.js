import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const propTypes = {
  /** Sets the Tab to active (adds active class)  */
  active: PropTypes.bool,
  /** CSS class of the Tab */
  className: PropTypes.string,
  /** Title of the Tab */
  title: PropTypes.string.isRequired,
  /** Function triggered on click event */
  onClick: PropTypes.func.isRequired,
};

const defaultProps = {
  active: false,
  className: 'tm-tab',
};

const Tab = ({ onClick, active, className, title }) => {
  return (
    <li className={className + (active ? ' tm-active' : '')}>
      <Button title={title} onClick={e => onClick(e)}>
        {title}
      </Button>
    </li>
  );
};

Tab.propTypes = propTypes;
Tab.defaultProps = defaultProps;

export default Tab;
