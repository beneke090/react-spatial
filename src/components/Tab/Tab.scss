.tm-tabs-item-button {
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid #eee;

  &:hover {
    background-color: $gray-lighter;
  }
}

.tm-active {
  color: $brand-primary;
  border-bottom: none;
  outline: none;
}
