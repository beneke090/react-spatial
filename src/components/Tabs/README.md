#

This demonstrates the use of Tabs

```jsx
import React, { useState } from 'react';
import Tabs from 'react-spatial/components/Tabs';
import TabItem from 'react-spatial/components/TabItem';



function TabsExample() {
  const [active, setActive] = useState(false);

  const activate = index => {
    console.log(index);
  };

  return (
    <div className="tm-tabs-example">
        <Tabs>
          <TabItem
          active = {active}
          title="Tab 1"
          onClick={() => activate('1')}
          >
          <p>Lorem Ipsum dolor sic amet</p>
          </TabItem>
          <TabItem
          active = {active}
          title="Tab 2"
          onClick= {() => activate('2')}
          />
        </Tabs>
    </div>
  );
}

<TabsExample />;
```
