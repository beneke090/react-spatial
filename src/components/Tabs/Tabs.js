import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /** CSS class of the container  */
  className: PropTypes.string,

  /** CSS class of the list of tabs  */
  classNameList: PropTypes.string,

  /** CSS class of the content  */
  classNameContent: PropTypes.string,

  /** Children content of the Tabs container.  */
  children: PropTypes.node.isRequired,
};

const defaultProps = {
  className: 'tm-tabs',
  classNameList: 'tm-tabs-list',
  classNameContent: 'tm-tabs-content',
};

function Tabs({ children, className, classNameList, classNameContent }) {
  return (
    <div className={className}>
      <ol className={classNameList}>{children}</ol>
      <div className={classNameContent}>
        {children.map(child => {
          if (!child.props.active) return null;
          return child.props.children;
        })}
      </div>
    </div>
  );
}

Tabs.propTypes = propTypes;
Tabs.defaultProps = defaultProps;

export default Tabs;
