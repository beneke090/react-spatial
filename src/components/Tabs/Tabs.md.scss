.tm-tabs-example {
  position: relative;
  height: 250px;
  width: 500px;

  .tm-tabs {
    position: absolute;
    width: 400px;
  }

  .tm-tabs-content {
    text-align: justify;
    height: 150px;

    p {
      margin: 0;
    }
  }
}
