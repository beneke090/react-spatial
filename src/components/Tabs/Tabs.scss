.tm-tabs {
  min-height: 50px;

  ol {
    margin: 0;
  }

  .tm-tab,
  .tm-tabs-content {
    border: 1px solid #eee;
  }

  .tm-tab {
    color: $gray;

    &.tm-active {
      color: $brand-primary;
      border-bottom: none;
    }
  }

  .tm-tabs-list {
    display: flex;
    list-style: none;
    padding: 0;

    li {
      flex-grow: 1;
    }
  }

  .tm-tabs-content {
    padding: 20px;
    border-top: none;
    overflow: auto;
  }
}
