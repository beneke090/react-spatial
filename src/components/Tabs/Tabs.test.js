import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Tabs from './Tabs';
import Tab from '../Tab';

configure({ adapter: new Adapter() });

describe('Tabs', () => {
  describe('should match snapshot.', () => {
    test('with two tabs.', () => {
      const component = renderer.create(
        <Tabs>
          <Tab title="Tab 1" onClick={() => {}}>
            Tab 1
          </Tab>
          <Tab active title="Tab 2" onClick={() => {}}>
            Tab 2
          </Tab>
        </Tabs>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
