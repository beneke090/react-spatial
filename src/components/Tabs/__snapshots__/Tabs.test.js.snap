// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should match snapshot. with two tabs. 1`] = `
<div
  className="tm-tabs"
>
  <ol
    className="tm-tabs-item-list"
  >
    <li
      className="tm-tabs-item"
    >
      <div
        aria-label="Tab 1"
        className="tm-tabs-item-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Tab 1"
      >
        Tab 1
      </div>
    </li>
    <li
      className="tm-tabs-item"
    >
      <div
        aria-label="Tab 2"
        className="tm-tabs-item-button tm-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title="Tab 2"
      >
        Tab 2
      </div>
    </li>
  </ol>
  <div
    className="tm-tabs-item-content"
  >
    Tab 2
  </div>
</div>
`;
